pkg/fieldfilters/fields.go

```diff
@@ -189,24 +204,22 @@ func (f *FieldFilter) Filter(event *tetragon.GetEventsResponse) (*tetragon.GetEv
 		}
 	}
 
-	rft := event.ProtoReflect()
-	rft.Range(func(eventDesc protoreflect.FieldDescriptor, _ protoreflect.Value) bool {
-		if eventDesc.ContainingOneof() == nil || !rft.Has(eventDesc) {
+	src := event.ProtoReflect()
+	dst := src.New()
+	var filterErr error
+	src.Range(func(fd protoreflect.FieldDescriptor, v protoreflect.Value) bool {
+		if fd.ContainingOneof() == nil || !src.Has(fd) {
 			return true
 		}
-		event := rft.Mutable(eventDesc).Message().Interface()
-		switch f.action {
-		case tetragon.FieldFilterAction_INCLUDE:
-			f.fields.Filter(event)
-		default:
-			f.fields.Prune(event)
-		}
+		event := src.Get(fd).Message().Interface()
+		dstEvent := dst.Mutable(fd).Message().Interface()
+		filterErr = fieldmask_utils.StructToStruct(f.fields, event, dstEvent)
```

Comment:
maybe erorrs.Join() is better here so that we don't miss errors? I believe this code will only keep the value from the last iteration.

Reply:
the intention was actually to return early (which I failed to do here) but I like your idea better